scripts/build-testing-sdk.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * This script bundles some scripts used for testing:
+ * - amd loader
+ * - polyfills
+ *   + fetch
+ *   + promises
+ *   + bace64 encoding
+ * - then deps for the testing framework
+ */
+const fs = require('fs');
+
+const sourceModules = getSourceFromModules([
+  // umd module loader
+  'loader.js',
+  // original testing sdk deps
+  'route-recognizer',
+  'fake-xml-http-request',
+  'pretender'
+]);
+
+const outSource = `
+${sourceModules[0]}
+(function (self) {
+;self.fetch = null;
+${sourceModules.slice(1).join('\n')}
+}(window));
+`;
+
+console.log(outSource);
+
+function getSourceFromModules(modules) {
```

Comment:
@minasmart is that a rule? I started doing this so that it's easier to walk through code. It allows for the main application logic to exist at the top of the page and supporting functions to be at the bottom. This means that if you have to walk through the code you see the major functionality first and then discover the supporting functions later. Should be a non-issue due to hoisting. But... If it's a rule with no discussion I will follow suite.